### ISSUE TITLE
Add a sample app that exercises all aspects of Knative

### ISSUE THREAD

**Issue by omerbensaadon** [knative/ux#33]
@skonto commented on [Thu Oct 15 2020](https://github.com/knative/docs/issues/2934)

**Describe the change you'd like to see**
Istio has this nice [sample app](https://istio.io/latest/docs/examples/bookinfo/) that demonstrates how to use it.
It would be nice to have a non-trivial example to demonstrate similar concepts in Knative. Not sure if there is any but for example for Eventing I didn't find anything beyond hello-wrold like samples..
I would suggest an app that is aligned with common Knative use cases eg. an IoT app that processes sensor readings using a microservices based approach. It could use a different language per microservice.

/cc @matzew @slinkydeveloper @abrennan89 WDYTH?

---

@github-actions[bot] commented on [Wed Jan 13 2021](https://github.com/knative/docs/issues/2934#issuecomment-759348661)

This issue is stale because it has been open for 90 days with no
activity. It will automatically close after 30 more days of
inactivity. Reopen the issue with `/reopen`.Mark the issue as
fresh by adding the comment `/remove-lifecycle stale`.

---

@abrennan89 commented on [Wed Jan 13 2021](https://github.com/knative/docs/issues/2934#issuecomment-759538795)

@omerbensaadon this one may interest you?

---

@omerbensaadon commented on [Thu Jan 14 2021](https://github.com/knative/docs/issues/2934#issuecomment-760253903)

@abrennan89 totally, this feels like a deliverable that ought to come out of the UX WG and is currently coming to light in User Interviews. 

CC: @csantanapr

---

@omerbensaadon commented on [Thu Jan 14 2021](https://github.com/knative/docs/issues/2934#issuecomment-760254485)

Once we wrap up this cycle of interviews (I think next week?) we may want to take a crack at creating an epic and splitting this issue into a few smaller issues to be worked.

---

@slinkydeveloper commented on [Thu Jan 14 2021](https://github.com/knative/docs/issues/2934#issuecomment-760268789)

That's nice!

---

@skonto commented on [Thu Jan 14 2021](https://github.com/knative/docs/issues/2934#issuecomment-760346232)

@omerbensaadon let me know if I can help there.

---

@pierDipi commented on [Wed Apr 14 2021](https://github.com/knative/docs/issues/2934#issuecomment-819801231)

@omerbensaadon any updates?

---

@skonto commented on [Thu Apr 15 2021](https://github.com/knative/docs/issues/2934#issuecomment-820210543)

@pierDipi @omerbensaadon We can create a proposal for the app to get started. I already have one.

---

@pierDipi commented on [Thu Apr 15 2021](https://github.com/knative/docs/issues/2934#issuecomment-820231204)

Cool, go ahead then.

---

@csantanapr commented on [Thu Apr 15 2021](https://github.com/knative/docs/issues/2934#issuecomment-820355136)

This is great a sample app(s) to explain Knative concepts is something I have being wanted to have Knative for a while. 

Is great to have some helping hands on this now. 

Would you like to write a proposal and present to the UX WG?

Someone from VMware that @omerbensaadon brought in is also working on an app. We should combine ideas and efforts. 

---

@omerbensaadon commented on [Thu Apr 15 2021](https://github.com/knative/docs/issues/2934#issuecomment-820408857)

sorry folks, let this one slip. 

@skonto would love to see a proposal! 

After you write the proposal, we can see where the work we have been doing with the Tanzu Developer's Center could fit together? 

Moving this issue to the UX Repo...

**Comment by omerbensaadon** [comment]
@skonto do you mind linking to your sample app if you've got the Github Link handy?

**Comment by omerbensaadon** [comment]
@skonto 
>Cool! I am working on it next week will have something for review, I recently developed a [small one](https://github.com/skonto/serverless-zero-demo-app) (not polished) and spent some time thinking about it anyway.

Looks like this is OpenShift-specific? I am assuming you mean a general Knative Sample App?

Check out the "Getting Started Experience" we've been working on https://julz.github.io/docs/dev/getting-started/getting-started/

Sample app could be a good candidate for "Next Steps"

**Comment by skonto** [comment]
Yes the idea is to move to a Knative one and more generic as most vendors to do to cover specific use cases eg. proposing a reference architecture etc.

**Comment by omerbensaadon** [comment]
We are currently working with @BrianMMcClain to create a sample app which leverages a Github issue as a database! 

Our outline of concepts is as follows: 

Begin Serving Intro
Use Konk.dev to install Knative
[Service + Source to URL] 
[Auto Scaling] 
[Revisions] 
[Traffic Splitting] 
FIN Serving Intro

Begin Eventing Intro
[Cloud Events]
[Producers + Consumers] 
[Broker + Trigger] 

FIN Eventing Intro

**Comment by skonto** [comment]
Ok nice in general it does not have to be just one sample app but a collection of standalone examples.
Some examples could be more complex and go beyond the getting started experience (depends on the persona),
so let's discuss in the proposal what we want to cover (let's not omit any current work for sure) ;)

**Comment by omerbensaadon** [comment]
I await your draft!

Agreed on multiple sample apps for different personas.

**Comment by pierDipi** [comment]
Thanks for creating the document, I like the direction :), ping us on this issue when it's ready.

**Comment by skonto** [comment]
@pierDipi @omerbensaadon I think we have enough use cases to start with. Should we discuss at the next UX wg meeting?

**Comment by skonto** [comment]
Added to the next meeting agenda.

**Comment by omerbensaadon** [comment]
Excited to chat about this today!

**Comment by omerbensaadon** [comment]
Removed "all" since that's a bit ambitious, having a number of these would be nice.

**Comment by Leo6Leo** [comment]
Here are some draft proposal for this issue:
[Knative Bookstore Application](https://docs.google.com/document/d/1dYvNdHFf4qoiIbWWOWMLAXDAFpEjjRsa9lP27tpqPTI/edit#heading=h.1d69849dwava)
[Sample IoT Application: Smart Environment Monitoring System](https://docs.google.com/document/d/11SEDYqW3UACecMyKNl_CnnwG9ljj1i18rbD7RBnaXw4/edit)

**Comment by Cali0707** [comment]
@Leo6Leo can you add this to the agenda for tomorrow's WG meeting?
